ConsoleApp1/ConsoleApp1/Program.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace TeleprompterConsole
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            RunTeleprompter().Wait();
+        }
+        private static async Task ShowTeleprompter(TelePrompterConfig config)
+        {
+            var words = ReadFrom("sampleQuotes.txt");
+            foreach (var word in words)
+            {
+                Console.Write(word);
+                if (!string.IsNullOrWhiteSpace(word))
+                {
+                    await Task.Delay(config.DelayInMilliseconds);
+                }
+            }
+            config.SetDone();
+        }
+
+        /// <summary>
+        /// Reads from file and returns word by word
+        /// </summary>
+        /// <param name="file">File being read from</param>
+        /// <returns>string for each word (or new line)</returns>
+        static IEnumerable<string> ReadFrom(string file)
+        {
+            string line;
+            using (var reader = File.OpenText(file))
+            {
+                while ((line = reader.ReadLine()) != null)
+                {
+                    var words = line.Split(' ');
```

Comment:
You'll have fun in QBCS or Unicode files. 

If you just call:

```cs
string[] ssize = myStr.Split(null); //Or myStr.Split()
```
or:

```cs
string[] ssize = myStr.Split(new char[0]);
```

then white-space is assumed to be the splitting character. From the string.Split(char[]) method's documentation page.

If the separator parameter is null or contains no characters, white-space characters are assumed to be the delimiters. White-space characters are defined by the Unicode standard and return true if they are passed to the Char.IsWhiteSpace method.

Reply:
OK - looked at doc page.  Bit more to it than that.  If you have other parameters - ie one of the other overrides, you need to specify type of null I think eg. default(char[]) or (char[])null and these are preferred to new char[0] as they do not require creation of a new object (or so I'm told via stackoverflow).  Correct me if I'm wrong